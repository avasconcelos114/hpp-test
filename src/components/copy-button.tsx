import React, { useEffect, useState } from 'react';

import { Button } from '@/components/ui/button';

const COPY_RESET_TIME = 5000;

export function CopyButton({
  text,
  testId,
  ariaLabel,
}: {
  text: string;
  testId: string;
  ariaLabel: string;
}) {
  const [isCopied, setIsCopied] = useState(false);

  const handleCopy = () => {
    navigator.clipboard.writeText(text);
    setIsCopied(true);
  };

  useEffect(() => {
    if (isCopied) {
      setTimeout(() => {
        setIsCopied(false);
      }, COPY_RESET_TIME);
    }
  }, [isCopied]);

  return (
    <Button
      variant='ghost'
      onClick={handleCopy}
      className='text-bvnk-primary h-[22px] p-0'
      data-testid={`copy-button-${testId}`}
      aria-label={ariaLabel}
      aria-pressed={isCopied}
    >
      {isCopied ? 'Copied!' : 'Copy'}
    </Button>
  );
}
