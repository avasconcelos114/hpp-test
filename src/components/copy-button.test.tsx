import React from 'react';
import { fireEvent, render, act } from '@testing-library/react';
import { describe, it, expect, vi } from 'vitest';
import { CopyButton } from './copy-button';

Object.assign(navigator, {
  clipboard: {
    writeText: vi.fn(),
  },
});

describe('CopyButton', () => {
  it('should render', () => {
    const screen = render(
      <CopyButton text='test' testId='test' ariaLabel='test' />,
    );
    expect(screen.getByTestId('copy-button-test')).toMatchSnapshot();
  });

  it('should copy text to clipboard', () => {
    const screen = render(
      <CopyButton text='test' testId='test' ariaLabel='test' />,
    );
    const button = screen.getByTestId('copy-button-test');
    fireEvent.click(button);
    expect(button.textContent).toBe('Copied!');
  });

  it('should reset text after timeout', async () => {
    vi.useFakeTimers();
    const screen = render(
      <CopyButton text='test' testId='test' ariaLabel='test' />,
    );
    const button = screen.getByTestId('copy-button-test');
    fireEvent.click(button);
    expect(button.textContent).toBe('Copied!');

    // Advance timers and flush React updates
    await act(async () => {
      vi.advanceTimersByTime(6000); // or whatever your timeout is
    });

    expect(button.textContent).toBe('Copy');
    vi.useRealTimers();
  });
});
