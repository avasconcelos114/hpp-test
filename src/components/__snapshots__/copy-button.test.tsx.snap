// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CopyButton > should render 1`] = `
<button
  aria-label="test"
  aria-pressed="false"
  class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-status-error/20 dark:aria-invalid:ring-status-error/40 aria-invalid:border-status-error cursor-pointer hover:bg-accent hover:text-accent-foreground dark:hover:bg-accent/50 has-[>svg]:px-3 text-bvnk-primary h-[22px] p-0"
  data-slot="button"
  data-testid="copy-button-test"
>
  Copy
</button>
`;
